merkle/tree_hasher.go

```diff
@@ -14,11 +14,25 @@
 
 package merkle
 
-// TreeHasher is the interface that the previous tree hasher struct implemented.
+// TreeHasher provides the hash functions needed to compute both chron and sparse merkle trees.
```

Comment:
I wrote chron to mean chronological - ie log trees, dense trees that are filled left to right. Is there a better term for this?